pkg/kubectl/describe.go

```diff
@@ -1765,6 +1765,7 @@ func (d *ServiceAccountDescriber) Describe(namespace, name string, describerSett
 	}
 
 	tokens := []api.Secret{}
+	existingTokens := sets.NewString()
 
 	tokenSelector := fields.SelectorFromSet(map[string]string{api.SecretTypeField: string(api.SecretTypeServiceAccountToken)})
 	options := api.ListOptions{FieldSelector: tokenSelector}
```

Comment:
This list is actually as expensive as listing all secrets in the namespace. Just do an unfiltered list here, and if err is nil, build the tokens list (adding a type check on each secret since we're no longer filtering the list), and build a "missingSecrets" string set

In error cases, missingSecrets will be empty

The only change to describeServiceAccount will be to take a missingSecrets string set and print "not found" after secrets with names in that set. All the other changes can be reverted